sklearn/ensemble/_forest.py

```diff
@@ -1120,6 +1120,7 @@ def _compute_partial_dependence_recursion(self, grid, target_features):
         averaged_predictions = np.zeros(
             shape=grid.shape[0], dtype=np.float64, order="C"
         )
+        target_features = np.asarray(target_features, dtype=np.intp, order="C")
```

Comment:
I believe only when necessary since it's asarray vs array. So copying depends on what is passed in. 

I followed the same pattern as 'grid' on L1119. 

If I don't add it, tho Cython does not do any automatic casting now that we've changed to 'intp_t' whereas I believe it was doing it for 'int'. This results in an error. 

Lmk if you think there's a better approach tho.